pkg/streamingpromql/compat/fallback_header.go

```diff
@@ -4,45 +4,25 @@ package compat
 
 import (
 	"context"
-	"net/http"
+	"fmt"
 
-	apierror "github.com/grafana/mimir/pkg/api/error"
+	"github.com/grafana/mimir/pkg/querier/api"
 )
 
-type engineFallbackContextKey int
-
-const forceFallbackEnabledContextKey = engineFallbackContextKey(0)
 const ForceFallbackHeaderName = "X-Mimir-Force-Prometheus-Engine"
 
-type EngineFallbackInjector struct{}
-
-func (i EngineFallbackInjector) Wrap(handler http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if value := r.Header.Get(ForceFallbackHeaderName); value != "" {
-			if value != "true" {
-				// Send a Prometheus API-style JSON error response.
-				w.Header().Set("Content-Type", "application/json")
-				w.WriteHeader(http.StatusBadRequest)
-				e := apierror.Newf(apierror.TypeBadData, "invalid value '%s' for '%s' header, must be exactly 'true' or not set", value, ForceFallbackHeaderName)
-
-				if body, err := e.EncodeJSON(); err == nil {
-					_, _ = w.Write(body)
-				}
-
-				return
-			}
-
-			r = r.WithContext(withForceFallbackEnabled(r.Context()))
-		}
-
-		handler.ServeHTTP(w, r)
-	})
+func ForceFallbackHeaderNameValidator(v string) error {
```

Comment:
nit: 

```suggestion
func ValidateForceFallbackHeaderName(v string) error {
```

to match `ValidateReadConsistency`